lib/package.gi

```diff
@@ -159,6 +159,172 @@ InstallGlobalFunction( SetPackageInfo, function( record )
     end );
 
 
+#############################################################################
+##
+#F  FindPackageInfosInSubdirectories( pkgdir, name )
+##
+##  Finds all PackageInfos in subdirectories of directory name in
+##  directory pkgdir, returns a list of their paths.
```

Comment:
I find this description difficult to understand. How about this:
> This function tests if <pkgdir> has a subdirectory named <name>`; if so, it iterates over all subdirectories
> of <name>, and returns a list of those which contain a PackageInfo.g files.

But thinking some more about it, I think it would be even better to replace this function by one that only takes one arguments, `path`, and which returns all subdirs of path that contain a PackageInfo.g. That is much easier to understand and explain.

Then, add another function which returns all subdirs of a path, if any; this function has more users, after all, even in this PR (in fact, even in this very function :-) ). It could look like this:
```gap
SubdirectoriesOfPath := function(path)
  local name, file, subs;
  subs := [];
  path := Directory(path);
  for name in Set(DirectoryContents(path)) do
    if name in [ ".", ".." ] then continue; fi;
    file := Filename(path, name);
    if IsDirectoryPath(file) then Add(subs, file); fi;
  od;
  return subs;
end;
```